### PR TITLE
[SPARK-44717][PYTHON][PS] Respect TimestampNTZ in resampling

### PR DESCRIPTION
**Pull request by HyukjinKwon**
### What changes were proposed in this pull request?

This PR proposes to respect `TimestampNTZ` type in resampling at pandas API on Spark.

### Why are the changes needed?

It still operates as if the timestamps are `TIMESTAMP_LTZ` even when `spark.sql.timestampType` is set to `TIMESTAMP_NTZ`, which is unexpected.

### Does this PR introduce _any_ user-facing change?

This fixes a bug so end users can use exactly same behaviour with pandas with `TimestampNTZType` - pandas does not respect the local timezone with DST. While we might need to follow this even for `TimestampType`, this PR does not address the case as it might be controversial.

### How was this patch tested?

Unittest was added.

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
cc @zhengruifeng and @attilapiros FYI

**Comment by HyukjinKwon** [comment]
cc @gengliangwang too

**Review by zhengruifeng — APPROVED** [review]

File: python/pyspark/pandas/tests/test_resample.py [hunk 1]
```diff
@@ -252,14 +254,32 @@ def test_dataframe_resample(self):
...
+    def check_series_resample(self):
         self._test_resample(self.pdf1.A, self.psdf1.A, ["4Y"], "right", None, "min")
         self._test_resample(self.pdf2.A, self.psdf2.A, ["13M"], "right", "left", "max")
         self._test_resample(self.pdf3.A, self.psdf3.A, ["1001H"], "right", "right", "sum")
         self._test_resample(self.pdf4.A, self.psdf4.A, ["6D"], None, None, "mean")
         self._test_resample(self.pdf5.A, self.psdf5.A, ["47T"], "left", "left", "var")
         self._test_resample(self.pdf6.A, self.psdf6.A, ["111S"], "right", "right", "std")
 
+    def test_series_resample(self):
+        self.check_series_resample()
```
Comment (attilapiros) [inline python/pyspark/pandas/tests/test_resample.py:266]: We are still depending on the TZ environment setting indirectly. So just by running the test on a different TZ this test (and also the test_dataframe_resample) would simply fail. 

We should either guarantee the correct TZ in the beginning of the test or validate the assumption and produce a meaningful error.

File: python/pyspark/pandas/tests/connect/test_parity_resample.py [hunk 2]
```diff
@@ -16,17 +16,25 @@
...
+class ResampleParityTests(
     ResampleTestsMixin, PandasOnSparkTestUtils, TestUtils, ReusedConnectTestCase
 ):
     pass
 
 
+class ResampleWithTimezoneTests(
+    ResampleWithTimezoneMixin, PandasOnSparkTestUtils, TestUtils, ReusedConnectTestCase
+):
+    @unittest.skip("SPARK-44731: Support 'spark.sql.timestampType' in Python Spark Connect client")
```
Comment (HyukjinKwon) [inline python/pyspark/pandas/tests/connect/test_parity_resample.py:33]: cc @ueshin FYI

**Comment by HyukjinKwon** [comment]
Merged to master and branch-3.5.

File: python/pyspark/pandas/tests/test_resample.py (hunk 1 above)
Comment (attilapiros) [reply python/pyspark/pandas/tests/test_resample.py]: @HyukjinKwon this is still an issue. The old tests still are failing when the TZ is not UTC i.e in America/New_York.

See https://github.com/attilapiros/spark/actions/runs/5884585704/job/15960073184?pr=5

Anyway for production we have the "spark.sql.timestampType=TIMESTAMP_NTZ" settings.

Comment (HyukjinKwon) [reply python/pyspark/pandas/tests/test_resample.py]: Yeah ... this PR enables `spark.sql.timestampType=TIMESTAMP_NTZ` as a workaround for now ..
To fix this, we need a bigger scope of change .. and can be arguable in a way.

**Pull request closed without merging**